mio/process/video.py

```diff
@@ -0,0 +1,429 @@
+                diff_video = NamedFrame(
+                    name=f"diff_{config.noise_patch.diff_multiply}x", video_frame=diff_frames
+                )
+                diff_video.export(
+                    output_dir / f"{pathstem}",
+                    suffix=True,
+                    fps=20,
+                )
+            if config.noise_patch.output_noise_patch:
+                noise_patch = NamedFrame(name="noise_patch", video_frame=noise_patchs)
+            if config.frequency_masking.output_mask:
+                freq_mask_frame = NamedFrame(
+                    name="freq_mask", static_frame=freq_mask * np.iinfo(np.uint8).max
+                )
+                freq_mask_frame.export(
+                    output_dir / f"{pathstem}",
+                    suffix=True,
+                    fps=20,
+                )
+
+            if config.frequency_masking.enable:
+                freq_domain_video = NamedFrame(name="freq_domain", video_frame=freq_domain_frames)
+                freq_filtered_video = NamedFrame(
+                    name="freq_filtered", video_frame=freq_filtered_frames
+                )
+                if config.frequency_masking.output_freq_domain:
+                    freq_domain_video.export(
+                        output_dir / f"{pathstem}",
+                        suffix=True,
+                        fps=20,
+                    )
+                if config.frequency_masking.output_result:
+                    freq_filtered_video.export(
+                        (output_dir / f"{pathstem}"),
+                        suffix=True,
+                        fps=20,
+                    )
+
+            min_proj_frame = NamedFrame(name="min_proj", static_frame=minimum_projection)
+
+            if config.interactive_display.enable:
+                videos = [
+                    raw_video,
+                    noise_patch,
+                    patched_video,
+                    freq_filtered_video,
+                    freq_domain_video,
+                    min_proj_frame,
+                    freq_mask_frame,
+                ]
+                VideoPlotter.show_video_with_controls(
+                    videos,
+                    start_frame=config.interactive_display.start_frame,
+                    end_frame=config.interactive_display.end_frame,
+                )
+
+    @staticmethod
+    def get_minimum_projection(image_list: list[np.ndarray]) -> np.ndarray:
+        """
+        Get the minimum projection of a list of images.
+
+        Parameters:
+        image_list (list[np.ndarray]): A list of images to project.
+
+        Returns:
+        np.ndarray: The minimum projection of the images.
+        """
+        stacked_images = np.stack(image_list, axis=0)
+        min_projection = np.min(stacked_images, axis=0)
+        return min_projection
+
+    @staticmethod
+    def normalize_video_stack(image_list: list[np.ndarray]) -> list[np.ndarray]:
```

Comment:
either should be a separate class like the other processor classes or a function, strange to have it here in a different form than all the other video processing methods.

Reply:
Done